src/Streaming/Internal/Produce.hs

```diff
@@ -156,3 +186,297 @@ untilRight mEither = Effect loop
         Right r -> Control.return $ Return r
 {-# INLINABLE untilRight #-}
 
+
+-- # The \"Affine\" 'Stream'
+-------------------------------------------------------------------------------
+
+-- | An *affine stream is represented with a state of type @x@,
+-- a possibly terminating step function of type @(x #-> m (Either (f x) r))@,
+-- and a stop-short function @(x #-> m r)@.
+--
+-- This mirrors the inverse unfold of a normal stream:
```

Comment:
inverse unfold?

Reply:
This is still here though, and still doesn't make much sense, can you explain to me what you meant?

Reply:
I didn't push the local changes I made. I was going to re-write it to just `unfold` and made the change in master. I originally meant the inverse of the `unfold` function.

Reply:
Ah, I see. It's a matter of point of view, I suppose. I guess that this is saying that `unfold` is an isomorphism. I would have just said that it mirrors the `unfold` function myself.